test/unit/ngsiv2/ngsiService/active-devices-test.js

```diff
@@ -256,6 +256,82 @@ describe('Active attributes test', function() {
         });
     });
 
+    describe('When the IoTA gets a set of values with a TimeInstant which are in ISO8601 format', function() {
+        var modifiedValues;
+
+        beforeEach(function(done) {
+
+            modifiedValues = [
+                {
+                    name: 'state',
+                    type: 'Boolean',
+                    value: 'true'
+                },
+                {
+                    name: 'TimeInstant',
+                    type: 'ISO8601',
+                    value: '2022-10-22T22:22:22Z'
+                }
+            ];
+
+            nock.cleanAll();
+
+            iotAgentConfig.timestamp = true;
+            iotAgentLib.activate(iotAgentConfig, done);
+        });
+
+        afterEach(function(done) {
+            delete iotAgentConfig.timestamp;
+            done();
+        });
+
+        it('should not fail', function(done) {
+            iotAgentLib.update('light1', 'Light', '', modifiedValues, function(error) {
+                should.not.exist(error);
+                contextBrokerMock.done();
+                done();
+            });
+        });
+    });
+
+    describe('When the IoTA gets a set of values with a TimeInstant which are in ISO8601 format with ms', function() {
+        var modifiedValues;
+
+        beforeEach(function(done) {
+
+            modifiedValues = [
+                {
+                    name: 'state',
+                    type: 'Boolean',
+                    value: 'true'
+                },
+                {
+                    name: 'TimeInstant',
+                    type: 'ISO8601',
+                    value: '2022-10-22T22:22:22.222Z'
+                }
+            ];
+
+            nock.cleanAll();
+
+            iotAgentConfig.timestamp = true;
+            iotAgentLib.activate(iotAgentConfig, done);
+        });
+
+        afterEach(function(done) {
+            delete iotAgentConfig.timestamp;
+            done();
+        });
+
+        it('should not fail', function(done) {
+            iotAgentLib.update('light1', 'Light', '', modifiedValues, function(error) {
+                should.not.exist(error);
+                contextBrokerMock.done();
+                done();
+            });
+        });
+    });
+
```

Comment:
The above unit test happens to be the same as existing:

```
    describe('When the IoT Agent receives new information and the timestamp flag is on', function() {
        var modifiedValues;

        beforeEach(function(done) {
            var time = new Date(1438760101468); // 2015-08-05T07:35:01.468+00:00

            modifiedValues = [
                {
                    name: 'state',
                    type: 'boolean',
                    value: true
                },
                {
                    name: 'dimming',
                    type: 'number',
                    value: 87
                }
            ];

            timekeeper.freeze(time);

            nock.cleanAll();

            contextBrokerMock = nock('http://192.168.1.1:1026')
                .matchHeader('fiware-service', 'smartGondor')
                .matchHeader('fiware-servicepath', 'gardens')
                .post('/v2/entities/light1/attrs',
                    utils.readExampleFile('./test/unit/ngsiv2/examples/contextRequests/updateContextTimestamp.json'))
                .reply(204);

            iotAgentConfig.timestamp = true;
            iotAgentLib.activate(iotAgentConfig, done);
        });

        afterEach(function(done) {
            delete iotAgentConfig.timestamp;
            timekeeper.reset();

            done();
        });
```

So I understand is redundant and should be removod from this PR. @dcalvoalonso what do you think?

Reply:
(While this is decided I have put the test in skip in 7c082ca3137d901d827bd20fda50894d73b9a275, so travis make work)

Reply:
Yes, for me it is redundant and must be removed. And sorry for not having helped before.

Reply:
Fixed in 9838184 (test removed)